### PR TITLE
Add ability to execute steps based on conditions

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: distribution/src/site/content/conditional-steps.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+<body>
+
+<h2>Conditional Steps</h2>
+
+<p>Conditional steps mechanism allows to choose a step to run depending on result
+of condition's evaluation. For example, let's consider the following steps:</p>
+
+<pre class="brush: java">
+    @Conditional(condition = TimeBasedNotification.class, value = "after_midnight")
+    @When("I send notificateion with text $text")
```
Comment (valfirst) [inline distribution/src/site/content/conditional-steps.html]: `notification`

File: distribution/src/site/content/conditional-steps.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    public void sendNotificationAfterMidnight(String text) {
+        // ...
+    }
+    
+    @Conditional(condition = TimeBasedNotification.class, value = "before_midnight")
+    @When("I send notificateion with text $text")
```
Comment (valfirst) [inline distribution/src/site/content/conditional-steps.html]: `notification`

File: distribution/src/site/content/conditional-steps.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+<p>Notes:</p>
+<ul>
+    <li>if a step is annotated as conditional then all the steps with the same wording must be conditioinal, otherwise exception is thrown</li>
+    <li>if few step's conditions match the first one is taken</li>
+    <li>if no step's conditions match then the step is marked as pending</li>
+</ul>
+
+</body>
+</html>
```
Comment (EDbarvinsky) [inline distribution/src/site/content/conditional-steps.html]: Please, add new line

File: jbehave-core/src/main/java/org/jbehave/core/condition/ConditionChecker.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package org.jbehave.core.condition;
+
+import java.util.function.Predicate;
+
+public interface ConditionChecker {
```
Comment (EDbarvinsky) [inline jbehave-core/src/main/java/org/jbehave/core/condition/ConditionChecker.java]: IConditionChecker for naming unification ?

**Comment by uarlouski** [comment]
not for behave

**Pull request merged**